### PR TITLE
test: Convert json-rpc e2e tests to TypeScript

### LINKED ISSUE #27698: Convert json-rpc e2e tests to TypeScrip
### What is this about?

Convert json-rpc e2e tests to TypeScrip

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by chloeYue**
## **Description**

- Convert e2e tests in `test/e2e/json-rpc/*` to TS
- Improve function `loginWithBalanceValidation()`

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27659?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/27698

## **Manual testing steps**

Tests pass on CI

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by chloeYue** [comment]
For some of the migrated tests, git detect it as a removed test and an added test, as i have already opened the PR, i will keep it as it is. For the next time, i will use the [git hint here](https://stackoverflow.com/questions/433111/how-to-make-git-mark-a-deleted-and-a-new-file-as-a-file-move) to make git recogniza them as modified tests but not removed and added tests before openning the PR during the development process.

**Review by HowardBraham — DISMISSED** [review]
> For some of the migrated tests, Git detected them as removed tests and added tests... next time, I will use the Git hint here to make Git recognize them as modified tests

Is there actually a foolproof way to do this?  The tricky thing is when converting to TS, you cannot rename and change in separate PRs.

**Comment by chloeYue** [comment]
> > For some of the migrated tests, Git detected them as removed tests and added tests... next time, I will use the Git hint here to make Git recognize them as modified tests
> 
> Is there actually a foolproof way to do this? The tricky thing is when converting to TS, you cannot rename and change in separate PRs.

That makes sense your comment. I'm not sure about this either. I tried it on this branch without success. I guess it might need to be done in when committing changes to force Git to recognize the file as a changed file during development. I will try next time to see if it works.

File: test/e2e/json-rpc/eth_call.spec.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-const {
-  withFixtures,
-  unlockWallet,
-  defaultGanacheOptions,
-} = require('../helpers');
-const { SMART_CONTRACTS } = require('../seeder/smart-contracts');
-const FixtureBuilder = require('../fixture-builder');
+import { strict as assert } from 'assert';
+import { keccak } from 'ethereumjs-util';
+import { withFixtures, defaultGanacheOptions } from '../helpers';
```
Comment (seaona) [inline test/e2e/json-rpc/eth_call.spec.ts:3]:
```suggestion
import { defaultGanacheOptions, withFixtures } from '../helpers';
```

File: test/e2e/json-rpc/eth_accounts.spec.ts [hunk 2]
```diff
@@ -1,10 +1,9 @@
-const { strict: assert } = require('assert');
-const {
-  withFixtures,
-  defaultGanacheOptions,
-  unlockWallet,
-} = require('../helpers');
-const FixtureBuilder = require('../fixture-builder');
+import { strict as assert } from 'assert';
+import { withFixtures, defaultGanacheOptions } from '../helpers';
```
Comment (seaona) [inline test/e2e/json-rpc/eth_accounts.spec.ts:2]:
```suggestion
import { defaultGanacheOptions, withFixtures } from '../helpers';
```

File: test/e2e/json-rpc/eth_chainId.spec.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { strict as assert } from 'assert';
+import { withFixtures, defaultGanacheOptions } from '../helpers';
```
Comment (seaona) [inline test/e2e/json-rpc/eth_chainId.spec.ts:2]:
```suggestion
import { defaultGanacheOptions, withFixtures } from '../helpers';
```

File: test/e2e/json-rpc/eth_estimateGas.spec.ts [hunk 4]
```diff
@@ -1,10 +1,9 @@
-const { strict: assert } = require('assert');
-const {
-  withFixtures,
-  defaultGanacheOptions,
-  unlockWallet,
-} = require('../helpers');
-const FixtureBuilder = require('../fixture-builder');
+import { strict as assert } from 'assert';
+import { withFixtures, defaultGanacheOptions } from '../helpers';
```
Comment (seaona) [inline test/e2e/json-rpc/eth_estimateGas.spec.ts:2]:
```suggestion
import { defaultGanacheOptions, withFixtures } from '../helpers';
```

File: test/e2e/json-rpc/eth_gasPrice.spec.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import { strict as assert } from 'assert';
+import { withFixtures, defaultGanacheOptions } from '../helpers';
```
Comment (seaona) [inline test/e2e/json-rpc/eth_gasPrice.spec.ts:2]:
```suggestion
import { defaultGanacheOptions, withFixtures } from '../helpers';
```

File: test/e2e/json-rpc/eth_subscribe.spec.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
+import { strict as assert } from 'assert';
+import { withFixtures, defaultGanacheOptions } from '../helpers';
```
Comment (seaona) [inline test/e2e/json-rpc/eth_subscribe.spec.ts:2]:
```suggestion
import { defaultGanacheOptions, withFixtures } from '../helpers';
```

**Review by seaona — DISMISSED** [review]
Overall LGTM :+1:  just a couple of minor suggestions around ordering - not blockers

**Review by seaona — DISMISSED** [review]

**Review by seaona — APPROVED** [review]

**Review by vthomas13 — APPROVED** [review]

**Pull request merged**